bundles/org.eclipse.search/search/org/eclipse/search/internal/ui/text/TextSearchPage.java

```diff
@@ -481,22 +484,19 @@ public void createControl(Composite parent) {
 		PlatformUI.getWorkbench().getHelpSystem().setHelp(result, ISearchHelpContextIds.TEXT_SEARCH_PAGE);
 }
 
-	private boolean validateRegex() {
+private boolean validateRegex() {
+	SearchDecoration sDecoration = new SearchDecoration();
```

Comment:
Instantiate this only in the `catch` and also rename it to `searchDecoration`

Reply:
done